sdks/how-to/how-to-create-a-user-from-a-patient.md

```diff
@@ -0,0 +1,151 @@
+  .withCrypto(webcrypto as any)
+  .withAuthProcessByEmailId(authProcessId)
+  .withAuthProcessBySmsId(authProcessId)
+  .build()
+
+const { publicKey, privateKey } = await anonymousMedTechApi.cryptoApi.RSA.generateKeyPair()
+const publicKeyHex = ua2hex(await anonymousMedTechApi.cryptoApi.RSA.exportKey(publicKey, 'spki'))
+const privateKeyHex = ua2hex(await anonymousMedTechApi.cryptoApi.RSA.exportKey(privateKey, 'pkcs8'))
+
+await anonymousMedTechApi.authenticationApi.authenticateAndAskAccessToItsExistingData(
+  patientUsername,
+  patientToken,
+  async () => ({ privateKey: privateKeyHex, publicKey: publicKeyHex }),
+)
+```
+
+The `authenticateAndAskAccessToItsExistingData` method will set up the User private and public key, it will create a 
+long-lived authentication token, and will send a Notification to all the Healthcare Professionals that have a delegation 
+for the Patient to ask access to their data.
+
+### The Doctor Receives the Notification and Gives Back Access
+
+Once the Notification is sent, on the Doctor side you can use the MedTech SDK to filter his Notifications and get the one related to 
+the new User.
+
+<!-- file://code-samples/how-to/create-user-for-patient/index.mts snippet:doctor gets pending notifications-->
+```typescript
+const newNotifications = await api.notificationApi.getPendingNotifications()
+const patientNotification = newNotifications.filter(
+  (notification) =>
+    notification.type === NotificationTypeEnum.NEW_USER_OWN_DATA_ACCESS &&
+    notification.responsible === patient.id,
+)[0]
+```
+
+Then, you can change the status of the Notification to signal that the operation is being taken care of.
+
+<!-- file://code-samples/how-to/create-user-for-patient/index.mts snippet:notification set ongoing-->
+```typescript
+const ongoingStatusUpdate = await api.notificationApi.updateNotificationStatus(
+  patientNotification,
+  'ongoing',
+)
+```
+
+To allow the new User to access all their own data, you can use the `giveAccessToAllDataOf` method.
+
+<!-- file://code-samples/how-to/create-user-for-patient/index.mts snippet:data sharing-->
+```typescript
+const sharedData = await api.patientApi.giveAccessToAllDataOf(patient.id)
+```
+
+If the method runs successfully, it will return a report of all the shared objects:
+<details>
+    <summary>Output</summary>
+
+```typescript
+{
+  patient: "YOUR_PATIENT_OBJECT",
+  statuses: {
+    dataSamples: { success: true, error: null, modified: 1 },
+    healthcareElements: { success: true, error: null, modified: 0 },
+    patient: { success: true, error: null, modified: 0 }
+  }
+}
+
+```
+
+</details>
+
+Finally, you can update the Notification status again to signal that the operation was completed successfully.
+
+<!-- file://code-samples/how-to/create-user-for-patient/index.mts snippet:completed status-->
+```typescript
+const completedStatusUpdate = await api.notificationApi.updateNotificationStatus(
+  ongoingStatusUpdate,
+  'completed',
+)
+```
```

Comment:
How come the notifications are not marked as completed automatically?

Reply:
To avoid creating a dependency between the services. Also there could be different types of notifications, so in my opinion it does make sense to separate the notification management and the operations needed to complete a notification, especially if they will be modified in the future.